cpp/src/arrow/dataset/scanner.h

```diff
@@ -55,6 +55,7 @@ constexpr int64_t kDefaultBatchSize = 1 << 17;  // 128Ki rows
 constexpr int32_t kDefaultBatchReadahead = 16;
 constexpr int32_t kDefaultFragmentReadahead = 4;
 constexpr int32_t kDefaultBytesReadahead = 1 << 25;  // 32MiB
+constexpr int64_t kDefaultStartOffset = -1;
```

Comment:
Would `std::optional` be better here?

Reply:
all feasible, let's see what others think.

Reply:
Well, why not simply 0? Isn't starting from 0 the default behavior?

Reply:
> Well, why not simply 0? Isn't starting from 0 the default behavior?

because 0 is a correct value set. I am using -1 to indicate that this parameter is not set.